proto/regen/ecocredit/marketplace/v1beta1/types.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package regen.ecocredit.marketplace.v1beta1;
+
+import "google/protobuf/timestamp.proto";
+
+message Filter {
+  // or is a list of criteria for matching credit batches. A credit which
+  // matches this filter must match at least one of these criteria.
+  repeated Criteria or = 1;
+
+  // Criteria is a simple filter criteria for matching a credit batch.
+  message Criteria {
+    // selector specifies the primary selector criteria. Every criteria
+    // must specify a selector and every selector criteria must be based on
+    // an assertion by some authority such as a credit class creator, credit
+    // issuer, seller or curator.
+    Selector selector = 1;
+
+    // project_location can be specified in three levels of granularity:
+    // country, sub-national-code, or postal code. If just country is given,
+    // for instance "US" then any credits in the "US" will be matched even
+    // their project location is more specific, ex. "US-NY 12345". If
+    // a country, sub-national-code and postal code are all provided then
+    // only projects in that postal code will match.
+    string project_location = 2;
+
+    // start_date is the beginning of the period during which a credit batch
+    // was quantified and verified. If it is empty then there is no start date
+    // limit.
+    google.protobuf.Timestamp min_start_date = 3;
+
+    // end_date is the end of the period during which a credit batch was
+    // quantified and verified. If it is empty then there is no end date
+    // limit.
+    google.protobuf.Timestamp end_date = 4;
+  }
+}
+
+enum SelectorType {
+  SELECTOR_TYPE_UNDEFINED = 0;
+  SELECTOR_TYPE_CLASS = 1;
+  SELECTOR_TYPE_PROJECT = 2;
+  SELECTOR_TYPE_BATCH = 3;
+}
```

Comment:
The initial implementation for basket criteria includes credit type. Is there a reason we are not including it here? Also, credit batch was not included but I think it makes sense to include it. See [basket specification](https://github.com/regen-network/regen-ledger/pull/681/files#diff-fd02ad2e3fca1937bc3295c4fe32d55cbfcb1547d0d347697ec205b69120adb1R80-R89).

Reply:
Including batch in baskets you mean?

Credit type isn't including because it doesn't define any issuing authority and thus is not considered safe here.

These criteria need to be a bit simpler than the baskets because of performance concerns btw